x/programs/rust/wasmlanche_sdk/src/types.rs

```diff
@@ -122,4 +155,56 @@ impl Argument for Program {
     fn is_primitive(&self) -> bool {
         true
     }
+    fn from_bytes(bytes: &[u8]) -> Self {
+        let id: i64 = i64::from_bytes(bytes);
+        Self::from(id)
+    }
+}
+
+impl Argument for String {
+    fn as_bytes(&self) -> Cow<'_, [u8]> {
+        Cow::Owned(self.as_bytes().to_vec())
+    }
+    fn from_bytes(bytes: &[u8]) -> Self {
+        String::from_utf8_lossy(bytes).to_string()
+    }
+}
+
+// Represents a vector with types that implement the Argument trait.
+impl<T> Argument for Vec<T>
+where
+    T: Argument,
+{
+    // Construct a VecArg from bytes
+    fn from_bytes(bytes: &[u8]) -> Self {
+        let mut result_vec = Vec::new();
+
+        // Current byte we are reading from
+        let mut current_byte = 0;
+        let num_bytes = bytes.len();
+
+        // TODO: check logic on empty vec
+        while current_byte < num_bytes {
+            // Convert the bytes to the type T
+            let elem: T = T::from_bytes(&bytes[current_byte..]);
+            current_byte += elem.len();
+            // Add the element to the result vec
+            result_vec.push(elem);
+        }
+
+        result_vec
+    }
+
+    fn as_bytes(&self) -> Cow<'_, [u8]> {
+        if self.is_empty() {
+            return Cow::Owned(Vec::new());
+        }
+
+        // avoid reallocation
+        let mut bytes = Vec::with_capacity(self.len() * self[0].len());
+        for elem in self {
+            bytes.extend_from_slice(&elem.as_bytes());
+        }
+        Cow::Owned(bytes)
```

Comment:
Don't worry about re-allocating for now. We should either handle the case where the length of `T` varies, or we should force `T` to have a constant length with another trait-bound. In either case, the compiler should take care of minimizing allocations. 

```suggestion
        let bytes = self.iter().flat_map(|val| val.as_bytes()).collect();
        Cow::Owned(bytes)
```